omegaconf/listconfig.py

```diff
@@ -101,18 +102,25 @@ def _validate_set(self, key: Any, value: Any) -> None:
                     )
 
         target_type = self._metadata.element_type
+        if target_type in (Any, None):
+            return
+
         value_type = OmegaConf.get_type(value)
-        if is_structured_config(target_type):
-            if (
-                target_type is not None
-                and value_type is not None
-                and not issubclass(value_type, target_type)
-            ):
-                msg = (
-                    f"Invalid type assigned : {type_str(value_type)} is not a "
-                    f"subclass of {type_str(target_type)}. value: {value}"
-                )
-                raise ValidationError(msg)
+
+        needs_type_validation = is_structured_config(value_type) or isinstance(
+            value, ValueNode
+        )
+        if (
+            needs_type_validation
+            and value_type is not None
+            and not issubclass(value_type, target_type)
+        ):
+            msg = (
+                f"Invalid type assigned : {type_str(value_type)} is not a "
+                f"subclass of {type_str(target_type)}. value: {value}"
+            )
```

Comment:
I extracted this variable. Do you think I should leave a message explaining why we need `needs_type_validation` ?